ee/tabby-webserver/src/oauth/mod.rs

```diff
@@ -0,0 +1,37 @@
+use std::sync::Arc;
+use axum::{Json, Router, routing};
+use axum::extract::{Query, State};
+use axum::http::StatusCode;
+use axum::middleware::from_fn_with_state;
+use serde::Deserialize;
+use tracing::error;
+use crate::oauth::github::GithubOAuthResponse;
+use crate::schema::ServiceLocator;
+
+mod github;
+
+pub fn routes(locator: Arc<dyn ServiceLocator>) -> Router {
+    Router::new()
+        .route("/github/callback", routing::get(github_callback))
+        .with_state(locator)
+}
+
+#[derive(Deserialize)]
+struct GithubCallbackParams {
+    code: String,
+    state: Option<String>,
+}
+
+async fn github_callback(
+    State(state): State<Arc<dyn ServiceLocator>>,
```

Comment:
I feel that it might be better to just inject `DbConn` directly, instead of relying on auth service, on this particular route.

So you don't have to do a wrapper layer in AuthenticationService. (AuthenticationService's target is to decouple graphql api to their dependencies, while for other routes, i feel it's not necessary).

Reply:
I'll suggest let ServiceLocator returns `DbConn` directly, to avoid creating a dummy wrapper inside of AuthenticationService.